xcm/xcm-builder/src/universal_exports.rs

```diff
@@ -0,0 +1,368 @@
+		if let Some(payment) = maybe_payment {
+			cost.push(payment);
+		}
+		Ok((v, cost))
+	}
+
+	fn deliver(validation: Router::Ticket) -> Result<XcmHash, SendError> {
+		Router::deliver(validation)
+	}
+}
+
+/// Implementation of `SendXcm` which wraps the message inside an `ExportMessage` instruction
+/// and sends it to a destination known to be able to handle it.
+///
+/// The `ExportMessage` instruction on the bridge is paid for from the local chain's sovereign
+/// account on the bridge. The amount paid is determined through the `ExporterFor` trait.
+pub struct SovereignPaidRemoteExporter<Bridges, Router, UniversalLocation>(
+	PhantomData<(Bridges, Router, UniversalLocation)>,
+);
+impl<Bridges: ExporterFor, Router: SendXcm, UniversalLocation: Get<InteriorMultiLocation>> SendXcm
+	for SovereignPaidRemoteExporter<Bridges, Router, UniversalLocation>
+{
+	type Ticket = Router::Ticket;
+
+	fn validate(
+		dest: &mut Option<MultiLocation>,
+		xcm: &mut Option<Xcm<()>>,
+	) -> SendResult<Router::Ticket> {
+		let d = dest.as_ref().ok_or(MissingArgument)?.clone();
+		let devolved = ensure_is_remote(UniversalLocation::get(), d).map_err(|_| NotApplicable)?;
+		let (remote_network, remote_location) = devolved;
+
+		let xcm = xcm.take().ok_or(MissingArgument)?;
+		let (bridge, maybe_payment) =
+			Bridges::exporter_for(&remote_network, &remote_location, &xcm).ok_or(NotApplicable)?;
+
+		let local_from_bridge =
+			UniversalLocation::get().invert_target(&bridge).map_err(|_| Unroutable)?;
+		let export_instruction =
+			ExportMessage { network: remote_network, destination: remote_location, xcm };
+
+		let message = Xcm(if let Some(ref payment) = maybe_payment {
+			let fees = payment
+				.clone()
+				.reanchored(&bridge, UniversalLocation::get())
+				.map_err(|_| Unroutable)?;
+			vec![
+				WithdrawAsset(fees.clone().into()),
+				BuyExecution { fees, weight_limit: Unlimited },
+				export_instruction,
+				RefundSurplus,
+				DepositAsset { assets: All.into(), beneficiary: local_from_bridge },
+			]
+		} else {
+			vec![export_instruction]
+		});
+
+		// We then send a normal message to the bridge asking it to export the prepended
+		// message to the remote chain. This will only work if the bridge will do the message
+		// export for free. Common-good chains will typically be afforded this.
+		let (v, mut cost) = validate_send::<Router>(bridge, message)?;
+		if let Some(bridge_payment) = maybe_payment {
+			cost.push(bridge_payment);
+		}
+		Ok((v, cost))
+	}
+
+	fn deliver(ticket: Router::Ticket) -> Result<XcmHash, SendError> {
+		Router::deliver(ticket)
+	}
+}
+
+pub trait DispatchBlob {
+	/// Dispatches an incoming blob and returns the unexpectable weight consumed by the dispatch.
+	fn dispatch_blob(blob: Vec<u8>) -> Result<(), DispatchBlobError>;
+}
+
+pub trait HaulBlob {
+	/// Sends a blob over some point-to-point link. This will generally be implemented by a bridge.
+	fn haul_blob(blob: Vec<u8>);
```

Comment:
What about handling some "point-to-point" link error here?

```suggestion
	fn haul_blob(blob: Vec<u8>) -> Result<(), HaulBlobError>;
```

and maybe HaulBlobError could be translated to `v3:SendError::Transport(#[codec(skip)] &'static str)`

so the bellow `HaulBlobExporter` could be then like:
```
	fn deliver((blob, hash): (Vec<u8>, XcmHash)) -> Result<XcmHash, SendError> {
		Bridge::haul_blob(blob).map_err(|e| SendError::Transport(e.into()))?;
		Ok(hash)
	}
```